src/hotspot/cpu/s390/templateInterpreterGenerator_s390.cpp

```diff
@@ -1134,7 +1134,15 @@ void TemplateInterpreterGenerator::generate_fixed_frame(bool native_call) {
   __ z_agr(Z_locals, Z_esp);
   // z_ijava_state->locals - i*BytesPerWord points to i-th Java local (i starts at 0)
   // z_ijava_state->locals = Z_esp + parameter_count bytes
-  __ z_stg(Z_locals, _z_ijava_state_neg(locals), fp);
+
+  __ z_ldgr(Z_F1, Z_R1); // preserve Z_R1, holding cache offset
```

Comment:
How expensive is this?
I remember those GPR <-> FPR transfers to be convenient, but inefficient. But that my have changed.
Would Z_R0 be available or is it occupied as well?

Reply:
updated.